### DISCUSSION TITLE
Show what files/directories match the exclude patterns

### DISCUSSION THREAD

**Discussion by jose1711** [borgbase/vorta#980] [Ideas]
I believe a cool feature - especially when configuring the backup for the first time - would be a way to show which files/dirs would be included and which would be excluded from the backup, together with size. Currently you see the total size of the folder to be backed up which gives a quick information but as soon as you start adding excludes its becoming less relevant.

Let's assume a scenario in which there's no space to make a local test backup and you'd like to have a general idea on how much the repository before compression/deduplication would take.

**Comment by m3nu** [comment]
Cool idea, but not easy to get right for all situations.

I see some overlap with code in #961. Maybe worth considering, @freder.

There is also overlap with #907.

**Reply by jose1711** [reply]
This is a poor-man's implementation which can help with constructing pattern list for the time being:
```
# list excluded files/dirs
tmp=$(mktemp -d); borg init -e none $tmp; borg create --dry-run --list --exclude PATTERN $tmp::test /source 2>&1 | grep ^x
```
